src/core/store/interface.ts

```diff
@@ -4,6 +4,34 @@ import { NotImplementedError } from "../utilities/errors";
 export type File = Buffer | string;
 export type Content = string | Buffer;
 
+export interface ReadWriteStream extends WriteStream, ReadStream {}
+
+/**
+ * WriteStream: a write stream intended writing a batch file
+ */
+export interface WriteStream {
+  write(chunk: unknown, encoding?: string, callback?: (error: Error | null | undefined) => void): boolean;
+  write(chunk: unknown, cb?: (error: Error | null | undefined) => void): boolean;
+  end(chunk: unknown, cb?: () => void): void;
+  end(chunk: unknown, encoding?: string, cb?: () => void): void;
+}
+
+/**
+ * ReadStream: a read stream intended reading a batch file
+ */
+export interface ReadStream {
+  read(this: unknown, size: number): void;
+  end(chunk: unknown, cb?: () => void): void;
+  end(chunk: unknown, encoding?: string, cb?: () => void): void;
+}
+
+/**
+ * PassThroughCallback: a callback intended for receiving a write stream to create a batch file
+ */
+export interface PassThroughCallback {
+  (stream: ReadWriteStream): void;
```

Comment:
We await this, so I think it should be a Promise<void>
```suggestion
  (stream: ReadWriteStream): Promise<void>;
```